style_guide.rst

```diff
@@ -0,0 +1,393 @@
+(however, when that guide contradicts our guide, our guide will
+be considered the normative document).
+
+Code layout
+------------
+
+Indentation
+^^^^^^^^^^^
+
+You must follow one of the following two indent styles:
+
+* `K&R style <https://en.wikipedia.org/wiki/Indentation_style#K&R>`_ (you can use any of the variants listed on that page, except BSD KNF), or
+* `Allman style <https://en.wikipedia.org/wiki/Indent_style#Allman_style>`_
+
+Make sure you choose one indent style and use it **consistently**. Please note that you can use `astyle <http://astyle.sourceforge.net/>`_ to take existing code and
+convert it to one of the above indent styles::
+
+    astyle --style=kr -r "*.c"
+    astyle --style=allman -r "*.c"
+
+Regardless of the descriptions of K&R and Allman you may find online, *you must use 4 spaces per indentation level*. This is the
+indentation level used by default by astyle.
+
+**Never use "tab characters".**
+
+Note that you can configure most text editors to insert 4 spaces whenever you press the
+`tab key <https://en.wikipedia.org/wiki/Tab_key>`_. To check whether your editor is properly
+configured, try the following: press the tab key, and then press the left arrow key. If the
+cursor jumps to the beginning of the line, your editor inserted a single "tab character".
+If the cursor only goes back one space (and you have to press the left arrow key three more
+times to get to the start of the line), then the editor is correctly inserting four spaces.
+
+Whether to use `tabs or spaces <http://programmers.stackexchange.com/questions/57/tabs-versus-spaces-what-is-the-proper-indentation-character-for-everything-in-e>`_
+is one of the many "religious wars" within the programming community.
+So, you may run into people (even people here on campus) who will feel very strongly about
+how tabs are better than spaces, and they may even present to you a number of cogent arguments
+in favor of using tabs.
+
+Do not listen to these people. They are wrong and cannot be trusted.
+
+
+Maximum Line Length
+^^^^^^^^^^^^^^^^^^^
+
+Limit all lines to a maximum of 80 characters. This is not a hard limit,
+and breaking it occasionally is fine, as long as you don't do it repeateadly
+and always by a small number of additional characters. Also, remember that
+most syntactical elements in C are delimited by whitespace, and that
+newlines are considered whitespace. So, the following is correct C::
+
+    if (VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_LONG_CONDITION ||
+        ANOTHER_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_LONG_CONDITION ||
+        strlen(s) < 100)
+    {
+        /* ... */
+    }
+
+Blank Lines
+^^^^^^^^^^^
+
+Separate function definitions with two blank lines.
+
+Extra blank lines may be used (sparingly) to separate groups of
+related functions.
+
+Use blank lines in functions, sparingly, to indicate logical sections.
+
```

Comment:
My experience has been that graders prefer when you use blank lines fairly generously

Reply:
Changed to "Unless a function is very short, the body of the function should include blank lines functions to indicate logical sections (with at most one blank line separating each logical section)."

Do you think that makes it clear that there is actually an expectation that they'll use blank lines?

Reply:
Yep, I think that wording works well!